{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "37e4a061",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "import time\n",
    "\n",
    "sys.path.append(os.path.abspath(\"..\"))\n",
    "\n",
    "\n",
    "from dask import dataframe as dd\n",
    "from cluster_config.slurm_cluster import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ad442059",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-06-11 15:17:28,837 - tornado.application - ERROR - Exception in callback functools.partial(<bound method IOLoop._discard_future_result of <tornado.platform.asyncio.AsyncIOMainLoop object at 0x000001CBD24ACB00>>, <Task finished name='Task-8145' coro=<SpecCluster._correct_state_internal() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:352> exception=NotImplementedError()>)\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\natal\\AppData\\Roaming\\Python\\Python312\\site-packages\\tornado\\ioloop.py\", line 758, in _run_callback\n",
      "    ret = callback()\n",
      "          ^^^^^^^^^^\n",
      "  File \"C:\\Users\\natal\\AppData\\Roaming\\Python\\Python312\\site-packages\\tornado\\ioloop.py\", line 782, in _discard_future_result\n",
      "    future.result()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 396, in _correct_state_internal\n",
      "    await asyncio.gather(*worker_futs)\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\tasks.py\", line 684, in _wrap_awaitable\n",
      "    return await awaitable\n",
      "           ^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8181' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8176' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8197' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8187' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8192' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8182' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8177' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8188' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8198' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8193' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8183' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8178' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8174' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8189' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8194' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8184' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8179' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8190' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8195' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8185' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8180' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8175' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8196' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8186' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n",
      "Task exception was never retrieved\n",
      "future: <Task finished name='Task-8191' coro=<_wrap_awaitable() done, defined at c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py:124> exception=NotImplementedError()>\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 125, in _wrap_awaitable\n",
      "    return await aw\n",
      "           ^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\distributed\\deploy\\spec.py\", line 74, in _\n",
      "    await self.start()\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 426, in start\n",
      "    out = await self._submit_job(fn)\n",
      "          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 409, in _submit_job\n",
      "    return await self._call(shlex.split(self.submit_command) + [script_filename])\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask_jobqueue\\core.py\", line 503, in _call\n",
      "    proc = await create_subproc(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\subprocess.py\", line 224, in create_subprocess_exec\n",
      "    transport, protocol = await loop.subprocess_exec(\n",
      "                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 1756, in subprocess_exec\n",
      "    transport = await self._make_subprocess_transport(\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"c:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\asyncio\\base_events.py\", line 528, in _make_subprocess_transport\n",
      "    raise NotImplementedError\n",
      "NotImplementedError\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dashboard: http://192.168.1.113:8787/status\n"
     ]
    }
   ],
   "source": [
    "client = get_slurm_cluster()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0f7244a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Time to read data: 0.03 seconds\n"
     ]
    }
   ],
   "source": [
    "time_start = time.perf_counter()\n",
    "df = dd.read_parquet(\"../data/yellow_tripdata_2023-*.parquet\", blocksize=\"16MB\")\n",
    "time_elapsed = time.perf_counter() - time_start\n",
    "print(f'Time to read data: {time_elapsed:.2f} seconds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "551e4059",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "    <div style=\"width: 24px; height: 24px; background-color: #e1e1e1; border: 3px solid #9D9D9D; border-radius: 5px; position: absolute;\"> </div>\n",
       "    <div style=\"margin-left: 48px;\">\n",
       "        <h3 style=\"margin-bottom: 0px;\">Client</h3>\n",
       "        <p style=\"color: #9D9D9D; margin-bottom: 0px;\">Client-39f1b046-46c6-11f0-9420-ec2e98dc7904</p>\n",
       "        <table style=\"width: 100%; text-align: left;\">\n",
       "\n",
       "        <tr>\n",
       "        \n",
       "            <td style=\"text-align: left;\"><strong>Connection method:</strong> Cluster object</td>\n",
       "            <td style=\"text-align: left;\"><strong>Cluster type:</strong> dask_jobqueue.SLURMCluster</td>\n",
       "        \n",
       "        </tr>\n",
       "\n",
       "        \n",
       "            <tr>\n",
       "                <td style=\"text-align: left;\">\n",
       "                    <strong>Dashboard: </strong> <a href=\"http://192.168.1.113:8787/status\" target=\"_blank\">http://192.168.1.113:8787/status</a>\n",
       "                </td>\n",
       "                <td style=\"text-align: left;\"></td>\n",
       "            </tr>\n",
       "        \n",
       "\n",
       "        </table>\n",
       "\n",
       "        \n",
       "\n",
       "        \n",
       "            <details>\n",
       "            <summary style=\"margin-bottom: 20px;\"><h3 style=\"display: inline;\">Cluster Info</h3></summary>\n",
       "            <div class=\"jp-RenderedHTMLCommon jp-RenderedHTML jp-mod-trusted jp-OutputArea-output\">\n",
       "    <div style=\"width: 24px; height: 24px; background-color: #e1e1e1; border: 3px solid #9D9D9D; border-radius: 5px; position: absolute;\">\n",
       "    </div>\n",
       "    <div style=\"margin-left: 48px;\">\n",
       "        <h3 style=\"margin-bottom: 0px; margin-top: 0px;\">SLURMCluster</h3>\n",
       "        <p style=\"color: #9D9D9D; margin-bottom: 0px;\">fcb16d22</p>\n",
       "        <table style=\"width: 100%; text-align: left;\">\n",
       "            <tr>\n",
       "                <td style=\"text-align: left;\">\n",
       "                    <strong>Dashboard:</strong> <a href=\"http://192.168.1.113:8787/status\" target=\"_blank\">http://192.168.1.113:8787/status</a>\n",
       "                </td>\n",
       "                <td style=\"text-align: left;\">\n",
       "                    <strong>Workers:</strong> 0\n",
       "                </td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                <td style=\"text-align: left;\">\n",
       "                    <strong>Total threads:</strong> 0\n",
       "                </td>\n",
       "                <td style=\"text-align: left;\">\n",
       "                    <strong>Total memory:</strong> 0 B\n",
       "                </td>\n",
       "            </tr>\n",
       "            \n",
       "        </table>\n",
       "\n",
       "        <details>\n",
       "            <summary style=\"margin-bottom: 20px;\">\n",
       "                <h3 style=\"display: inline;\">Scheduler Info</h3>\n",
       "            </summary>\n",
       "\n",
       "            <div style=\"\">\n",
       "    <div>\n",
       "        <div style=\"width: 24px; height: 24px; background-color: #FFF7E5; border: 3px solid #FF6132; border-radius: 5px; position: absolute;\"> </div>\n",
       "        <div style=\"margin-left: 48px;\">\n",
       "            <h3 style=\"margin-bottom: 0px;\">Scheduler</h3>\n",
       "            <p style=\"color: #9D9D9D; margin-bottom: 0px;\">Scheduler-755502b0-8c02-4dfb-9d57-41142e6cc8de</p>\n",
       "            <table style=\"width: 100%; text-align: left;\">\n",
       "                <tr>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Comm:</strong> tcp://192.168.1.113:50525\n",
       "                    </td>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Workers:</strong> 0 \n",
       "                    </td>\n",
       "                </tr>\n",
       "                <tr>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Dashboard:</strong> <a href=\"http://192.168.1.113:8787/status\" target=\"_blank\">http://192.168.1.113:8787/status</a>\n",
       "                    </td>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Total threads:</strong> 0\n",
       "                    </td>\n",
       "                </tr>\n",
       "                <tr>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Started:</strong> Just now\n",
       "                    </td>\n",
       "                    <td style=\"text-align: left;\">\n",
       "                        <strong>Total memory:</strong> 0 B\n",
       "                    </td>\n",
       "                </tr>\n",
       "            </table>\n",
       "        </div>\n",
       "    </div>\n",
       "\n",
       "    <details style=\"margin-left: 48px;\">\n",
       "        <summary style=\"margin-bottom: 20px;\">\n",
       "            <h3 style=\"display: inline;\">Workers</h3>\n",
       "        </summary>\n",
       "\n",
       "        \n",
       "\n",
       "    </details>\n",
       "</div>\n",
       "\n",
       "        </details>\n",
       "    </div>\n",
       "</div>\n",
       "            </details>\n",
       "        \n",
       "\n",
       "    </div>\n",
       "</div>"
      ],
      "text/plain": [
       "<Client: 'tcp://192.168.1.113:50525' processes=0 threads=0, memory=0 B>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "client"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cb142207",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mKeyboardInterrupt\u001b[39m                         Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[10]\u001b[39m\u001b[32m, line 7\u001b[39m\n\u001b[32m      5\u001b[39m \u001b[38;5;66;03m# Grouping\u001b[39;00m\n\u001b[32m      6\u001b[39m time_start = time.perf_counter()\n\u001b[32m----> \u001b[39m\u001b[32m7\u001b[39m courses_per_month_and_year = \u001b[43mdf\u001b[49m\u001b[43m.\u001b[49m\u001b[43mgroupby\u001b[49m\u001b[43m(\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mmonth\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43myear\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43msize\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m      8\u001b[39m time_elapsed = time.perf_counter() - time_start\n\u001b[32m      9\u001b[39m \u001b[38;5;28mprint\u001b[39m(\u001b[33mf\u001b[39m\u001b[33m'\u001b[39m\u001b[33mTime to group data: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mtime_elapsed\u001b[38;5;132;01m:\u001b[39;00m\u001b[33m.2f\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m seconds\u001b[39m\u001b[33m'\u001b[39m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask\\base.py:373\u001b[39m, in \u001b[36mDaskMethodsMixin.compute\u001b[39m\u001b[34m(self, **kwargs)\u001b[39m\n\u001b[32m    349\u001b[39m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34mcompute\u001b[39m(\u001b[38;5;28mself\u001b[39m, **kwargs):\n\u001b[32m    350\u001b[39m \u001b[38;5;250m    \u001b[39m\u001b[33;03m\"\"\"Compute this dask collection\u001b[39;00m\n\u001b[32m    351\u001b[39m \n\u001b[32m    352\u001b[39m \u001b[33;03m    This turns a lazy Dask collection into its in-memory equivalent.\u001b[39;00m\n\u001b[32m   (...)\u001b[39m\u001b[32m    371\u001b[39m \u001b[33;03m    dask.compute\u001b[39;00m\n\u001b[32m    372\u001b[39m \u001b[33;03m    \"\"\"\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m373\u001b[39m     (result,) = \u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtraverse\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    374\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\site-packages\\dask\\base.py:681\u001b[39m, in \u001b[36mcompute\u001b[39m\u001b[34m(traverse, optimize_graph, scheduler, get, *args, **kwargs)\u001b[39m\n\u001b[32m    678\u001b[39m     expr = expr.optimize()\n\u001b[32m    679\u001b[39m     keys = \u001b[38;5;28mlist\u001b[39m(flatten(expr.__dask_keys__()))\n\u001b[32m--> \u001b[39m\u001b[32m681\u001b[39m     results = \u001b[43mschedule\u001b[49m\u001b[43m(\u001b[49m\u001b[43mexpr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    683\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m repack(results)\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\threading.py:655\u001b[39m, in \u001b[36mEvent.wait\u001b[39m\u001b[34m(self, timeout)\u001b[39m\n\u001b[32m    653\u001b[39m signaled = \u001b[38;5;28mself\u001b[39m._flag\n\u001b[32m    654\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m signaled:\n\u001b[32m--> \u001b[39m\u001b[32m655\u001b[39m     signaled = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_cond\u001b[49m\u001b[43m.\u001b[49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    656\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m signaled\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Users\\natal\\PycharmProjects\\Large-scale-dataset-processing-using-Dask\\.conda\\Lib\\threading.py:359\u001b[39m, in \u001b[36mCondition.wait\u001b[39m\u001b[34m(self, timeout)\u001b[39m\n\u001b[32m    357\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m    358\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m timeout > \u001b[32m0\u001b[39m:\n\u001b[32m--> \u001b[39m\u001b[32m359\u001b[39m         gotit = \u001b[43mwaiter\u001b[49m\u001b[43m.\u001b[49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    360\u001b[39m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m    361\u001b[39m         gotit = waiter.acquire(\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "\u001b[31mKeyboardInterrupt\u001b[39m: "
     ]
    }
   ],
   "source": [
    "df['month'] = df['tpep_pickup_datetime'].dt.month\n",
    "df['year'] = df['tpep_pickup_datetime'].dt.year\n",
    "df['trip_distance_km'] = df['trip_distance'] * 1.60934\n",
    "\n",
    "# Grouping\n",
    "time_start = time.perf_counter()\n",
    "courses_per_month_and_year = df.groupby(['month', 'year']).size().compute()\n",
    "time_elapsed = time.perf_counter() - time_start\n",
    "print(f'Time to group data: {time_elapsed:.2f} seconds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38bd25ae",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
